### PR TITLE
feat: USDC warp route config (Solana <> Eclipse)

### PR DESCRIPTION
**Pull request by daniel-savu**

### CODE REVIEW

File: rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "solanamainnet": {
+    "type": "collateral",
+    "decimals": 6,
```
Comment (daniel-savu) [inline rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json:4]: based on the solana and ethereum contracts:
- https://etherscan.io/token/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48
- https://solscan.io/token/EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v

File: rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    "interchainGasPaymaster": "AkeHBbE5JkwVppujCQQ6WuxsVsJtruBAjUo6fDCFp6fF",
+    "token": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+    "splTokenProgram": "token"
+  },
+  "eclipsemainnet": {
+    "type": "synthetic",
+    "decimals": 6,
```
Comment (daniel-savu) [inline rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json:11]: hoping that 6 here works, and the only constraint is that the upper limit is 9 decimals @tkporter?

File: rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "name": "USDC",
```
Comment (daniel-savu) [inline rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json:12]: should this be "Circle USD" instead?

File: rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json (hunk 2 above)
Comment (tkporter) [reply rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json:11]: yeah agree 6 works best given both Solana and Ethereum use it. and yes afaik 9 as an upper limit is the only constraint

File: rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json (hunk 3 above)
Comment (tkporter) [reply rust/sealevel/environments/mainnet3/warp-routes/eclipseusdc/token-config.json:12]: I'd probably do `USD Coin` to match what is on [Ethereum](https://etherscan.io/token/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48#readProxyContract#F16) and [Solana](https://solscan.io/token/EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v#metadata)

**Review by tkporter — APPROVED** [review]
nice

**Review by jmrossy — APPROVED** [review]

**Pull request merged**